### PR TITLE
Results refactor

### PR DESCRIPTION
**Pull request by pietgeursen**

### CODE REVIEW

File: position/sync/chooseOneResults.js [hunk 1]
```diff
@@ -1,4 +1,5 @@
-const isArray = require('isarray')
```
Comment (mixmix) [inline position/sync/chooseOneResults.js:1]: rm package `isarray`

File: test/poll/async/get.test.js [hunk 2]
```diff
@@ -49,11 +49,8 @@ test('pull.async.get', t => {
 
         t.equal(data.positions.length, 2, 'has positions')
 
-        t.deepEqual(data.results, {
-          1: [katie.id], // TODO update this data structure
-          2: [piet.id],
-          errors: [] // TODO prune this later
-        }, 'has results!')
+        t.ok(data.results.results[1].voters[katie.id])
+        t.ok(data.results.results[2].voters[piet.id])
```
Comment (mixmix) [inline test/poll/async/get.test.js]: API should be like : 

```
t.ok(data.results[1].voters[katie.id])
```

File: test/position/sync/chooseOneResults.test.js [hunk 3]
```diff
@@ -79,7 +104,7 @@ test('ChooseOneResults - A position stated after the closing time of the poll is
   ]
 
   const actual = chooseOneResults({positions, poll: validPoll})
-  t.false(actual[0], 'invalid vote is not counted')
```
Comment (mixmix) [inline test/position/sync/chooseOneResults.test.js:82]: add test for there being an error

Comment (pietgeursen) [reply test/position/sync/chooseOneResults.test.js:82]: heh, this is tested in the test below

**Pull request merged**